fluent.syntax/fluent/syntax/serializer.py

```diff
@@ -179,30 +156,39 @@ def serialize_placeable(placeable):
 
 def serialize_expression(expression):
     if isinstance(expression, ast.StringLiteral):
-        return '"{}"'.format(expression.raw)
+        return '"{}"'.format(expression.value)
     if isinstance(expression, ast.NumberLiteral):
         return expression.value
-    if isinstance(expression, ast.MessageReference):
-        return expression.id.name
-    if isinstance(expression, ast.FunctionReference):
-        return expression.id.name
-    if isinstance(expression, ast.TermReference):
-        return '-{}'.format(expression.id.name)
     if isinstance(expression, ast.VariableReference):
         return '${}'.format(expression.id.name)
-    if isinstance(expression, ast.AttributeExpression):
-        return serialize_attribute_expression(expression)
-    if isinstance(expression, ast.VariantExpression):
-        return serialize_variant_expression(expression)
-    if isinstance(expression, ast.CallExpression):
-        return serialize_call_expression(expression)
+    if isinstance(expression, ast.TermReference):
+        return serialize_reference_expression(
+            expression,
+            id='-{}'.format(expression.id.name)
+        )
+    if isinstance(
+        expression, (
+            ast.MessageReference,
+            ast.FunctionReference,
+        )
+    ):
+        return serialize_reference_expression(expression)
```

Comment:
This is a case each in the js impl. I opted for doing a joint `isinstance` here, but don't care much either way.